Lib/posixpath.py

```diff
@@ -432,7 +433,10 @@ def realpath(filename, *, strict=False):
         curdir = '.'
         pardir = '..'
         getcwd = os.getcwd
+    return _realpath(filename, strict, sep, curdir, pardir, getcwd)
 
+def _realpath(filename, strict=False, sep=sep, curdir=curdir, pardir=pardir,
+              getcwd=os.getcwd, lstat=os.lstat, readlink=os.readlink, maxlinks=None):
```

Comment:
Is this something for `genericpath`? It doesn't rely on anything of `posixpath`.

Reply:
It does a few somewhat POSIX-y things:

1. It resolves symlinks and `..` segments progressively, rather than resolving `..` segments upfront as Windows does
2. It supports only one path separator
3. It doesn't support drives (either DOS-style or UNC)

But I see your point. I'll think about it!

Reply:
Is it difficult to make the first difference an option like for coreutils?

Reply:
I think `realpath(normpath(p))` will accomplish that, so idk if there's much of a call for it.

Reply:
Symlinks can also include other symlinks followed by `..`.

Reply:
Symlink targets can themselves contain `..` segments. What does `realpath --logical` do about those?

Reply:
Yeah, that makes sense to me. I guess `realpath --logical` does a single pass of naively collapsing `..` segments in the original string, and then runs the normal path resolution process thereafter, and so `realpath(normpath(p))` should be equivalent in Python.